src/algebra/group/pi.lean

```diff
@@ -331,6 +331,42 @@ begin
   { simp [function.update_noteq h.symm, h] }
 end
 
+@[to_additive pi.single_add_single_eq_single_add_single]
+lemma pi.mul_single_mul_mul_single_eq_mul_single_mul_mul_single
+  {M : Type*} [comm_monoid M] {k l m n : I} {u v : M} (hu : u ≠ 1) (hv : v ≠ 1) :
+  mul_single k u * mul_single l v = mul_single m u * mul_single n v ↔
+  (k = m ∧ l = n) ∨ (u = v ∧ k = n ∧ l = m) ∨ (u * v = 1 ∧ k = l ∧ m = n) :=
+begin
+  refine ⟨λ h, _, _⟩,
+  { have hk := congr_fun h k,
+    have hl := congr_fun h l,
+    have hm := congr_fun h m,
+    have hn := congr_fun h n,
+    simp only [mul_apply, mul_single_apply, if_pos rfl] at hk hl hm hn,
+    by_cases hmk : m = k,
```

Comment:
You'll have a much easier time using `rcases eq_or_ne m k with rfl | hmk`, as this does the rewriting for you.